app/controllers/TaskController.scala

```diff
@@ -235,46 +236,103 @@ class TaskController @Inject()(annotationDAO: AnnotationDAO,
               params.editRotation,
               false
             )
-            .map(Some(_))
+            .map(v => Some((v, None)))
         } else Fox.successful(None)
       } yield volumeTracingOpt
     }
 
-  //Note that from-files tasks do not support volume tracings yet
-  @SuppressWarnings(Array("OptionGet")) //We surpress this warning because we know each skeleton exists due to `toSkeletonSuccessFox`
+  private def getTracingBases(skeletonBaseOpts: List[Option[SkeletonTracing]],
+                              volumeBaseOpts: List[Option[(VolumeTracing, Option[File])]],
+                              fullParams: List[TaskParameters],
+                              taskType: TaskType,
+                              organizationId: ObjectId)(
+      implicit ctx: DBAccessContext,
+      m: MessagesProvider): Fox[(List[Option[SkeletonTracing]], List[Option[(VolumeTracing, Option[File])]])] =
+    if (taskType.tracingType == TracingType.skeleton)
+      if (volumeBaseOpts.forall(_.isEmpty)) Fox.failure(Messages("taskType.mismatch", "skeleton", "volume"))
+      else Fox.successful((skeletonBaseOpts, volumeBaseOpts))
+    else if (taskType.tracingType == TracingType.volume)
+      if (skeletonBaseOpts.forall(_.isEmpty)) Fox.failure(Messages("taskType.mismatch", "volume", "skeleton"))
```

Comment:
this does not work for me as expected. could it be that this is the wrong way round? what exactly is expected for each tasktype?

Reply:
Yes, you are correct. I just changed it. Maybe you could have another look :slightly_smiling_face: